### PR TITLE
Create a deepcopy of config in api client

### PR DESCRIPTION
**Pull request by kartikgupta-db**
## Changes
* We do not want the config to change after API client has been initialised. One API client should be linked to 1 set of Config variables ONLY.

- [x] `make test` run locally
- [x] `make fmt` applied
- [x] relevant integration tests applied

### CODE REVIEW

File: databricks/sdk/__init__.py [hunk 1]
```diff
@@ -58,6 +60,11 @@
 
 
 class WorkspaceClient:
+    _config: client.Config
+
+    @property
+    def config(self):
+        return deepcopy(self._config)
```
Comment (nfx) [inline databricks/sdk/__init__.py:67]: i'm not sure it's a good thing. `w.config.authenticate()` should reuse the same refresh token

**Review by nfx — CHANGES_REQUESTED** [review]

File: databricks/sdk/__init__.py (hunk 1 above)
Comment (kartikgupta-db) [reply databricks/sdk/__init__.py:67]: Riggghttt. Completely forgot about this. We can atleast restrict the initial config form being passed around. If they directly reuse config from client, or edit it, atleast they know why stuff is changing.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from databricks.connect import DatabricksSession
+from databricks.sdk.core import Config
+
+config = Config(host='https://x', token='x', product="xapp", product_version="0.0.0")
```
Comment (nfx) [inline main.py:4]: please remove this file

**Review by nfx — CHANGES_REQUESTED** [review]
i'm still not sure that we should do it this way. if we want to prevent config from being modified, we can add these in Config:

```python
    self._init_auth()
    self._initialized = True
```

and these in `ConfigAttribute`:

```python
    def __set__(self, cfg: 'Config', value: any):
        if cfg._initialized: raise ValueError(...)
        cfg._inner[self.name] = self.transform(value)
```

and test in in DBR for dbr-native auth

**Review by nfx — APPROVED** [review]

**Pull request merged**